File: android/src/main/kotlin/com/fit321/fitui/components/FitProfileHeader.kt
package com.fit321.fitui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitSpacing

// ============================================================================
// FitProfileHeader — coach / athlete profile hero.
// Mirrors Swift FitProfileHeader. See `docs/components.md`
// § FitProfileHeader.
// ============================================================================

@Composable
fun FitProfileHeader(
    initials: String,
    name: String,
    sports: List<String> = emptyList(),
    location: String? = null,
    bio: String? = null,
    onEdit: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current
    val metadataParts = mutableListOf<String>()
    if (!location.isNullOrEmpty()) metadataParts.add(location)
    if (sports.isNotEmpty()) metadataParts.add(sports.joinToString(", "))
    val metadata = metadataParts.joinToString(" · ")

    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(FitSpacing.sp4),
        verticalArrangement = Arrangement.spacedBy(FitSpacing.sp4)
    ) {
        Row(
            verticalAlignment = Alignment.Top,
            horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp4)
        ) {
            FitAvatar(initials = initials, size = FitAvatarSize.Xl)

            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(FitSpacing.sp1)
            ) {
                Row(
                    verticalAlignment = Alignment.Top,
                    horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp2)
                ) {
                    Text(
                        name,
                        style = FitFont.heading1.copy(fontSize = 24.sp, fontWeight = FontWeight.SemiBold),
                        color = theme.textPrimary,
                        modifier = Modifier.weight(1f)
                    )
                    if (onEdit != null) {
                        FitIconBtn(
                            icon = Icons.Default.Edit,
                            color = FitIconBtnColor.Primary,
                            onClick = onEdit
                        )
                    }
                }
                if (metadata.isNotEmpty()) {
                    Text(metadata, style = FitFont.body2, color = theme.textTertiary)
                }
            }
        }

        if (!bio.isNullOrEmpty()) {
            Text(bio, style = FitFont.body1, color = theme.textSecondary)
        }
    }
}

File: Sources/FitUI/Components/FitProfileHeader.swift
import SwiftUI

// MARK: - FitProfileHeader
//
// Profile / coach hero — 80×80 avatar + name + metadata row +
// optional bio, optional trailing edit pencil icon. Used at the top
// of profile screens (own profile, viewed coach profile).
// See `docs/components.md` § FitProfileHeader.

public struct FitProfileHeader: View {
    let initials: String
    let imageURL: URL?
    let name: String
    let sports: [String]
    let location: String?
    let bio: String?
    let onEdit: (() -> Void)?

    @Environment(\.fitTheme) private var theme

    public init(
        initials: String,
        imageURL: URL? = nil,
        name: String,
        sports: [String] = [],
        location: String? = nil,
        bio: String? = nil,
        onEdit: (() -> Void)? = nil
    ) {
        self.initials = initials
        self.imageURL = imageURL
        self.name = name
        self.sports = sports
        self.location = location
        self.bio = bio
        self.onEdit = onEdit
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: FitSpacing.sp4) {
            HStack(alignment: .top, spacing: FitSpacing.sp4) {
                FitAvatar(initials: initials, size: .xl, image: imageURL)

                VStack(alignment: .leading, spacing: FitSpacing.sp1) {
                    HStack(alignment: .top, spacing: FitSpacing.sp2) {
                        Text(name)
                            .font(.custom(FitFont.family, size: 24).weight(.semibold))
                            .foregroundColor(theme.textPrimary)
                            .frame(maxWidth: .infinity, alignment: .leading)

                        if let onEdit {
                            FitIconBtn(systemName: "pencil", color: .primary, action: onEdit)
                        }
                    }

                    if !metadata.isEmpty {
                        Text(metadata)
                            .font(FitFont.body2)
                            .foregroundColor(theme.textTertiary)
                    }
                }
            }

            if let bio, !bio.isEmpty {
                Text(bio)
                    .font(FitFont.body1)
                    .foregroundColor(theme.textSecondary)
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
        .padding(FitSpacing.sp4)
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private var metadata: String {
        var parts: [String] = []
        if let location, !location.isEmpty { parts.append(location) }
        if !sports.isEmpty { parts.append(sports.joined(separator: ", ")) }
        return parts.joined(separator: " · ")
    }
}
